### ISSUE TITLE
Local scope in REPL

### ISSUE THREAD

**Issue by ivan-gavran** [informalsystems/quint#464]
I am experimenting with the REPL, and encountered a curious example: having defined `val x = 0`, upon trying to use `x` in a local context, e.g. `0.to(100).exists(x => fahrenheit(x) == x)`, I get 

>  error: Conflicting definitions found for name x

**Comment by bugarela** [comment]
Hey, thanks for reporting! That's actually a shadowing problem, as quint doesn't allow shadowing and identifies it as a conflict. Try switching your lambda parameter to some other name:
```
0.to(100).exists(y => fahrenheit(x) == y)
```
or the other way around, I'm not sure which way the conversion is going here:
```
0.to(100).exists(y => fahrenheit(y) == x)
```

**Comment by bugarela** [comment]
The error reporting for this could be clearer, tho. The highlighting currently looks like this:
```
>>> 0.to(100).exists(x => x == x)
syntax error: <input>:0:1 - error: Conflicting definitions found for name x in module __REPL
0: val x = 1
   ^^^^^^^^^
<input>:2:18 - error: Conflicting definitions found for name x in module __REPL
2: 0.to(100).exists(x => x == x)
                    ^^^^^^^^^^^
```
                    
where it should be

```
>>> 0.to(100).exists(x => x == x)
syntax error: <input>:0:1 - error: Conflicting definitions found for name x in module __REPL
0: val x = 1
       ^    
<input>:2:18 - error: Conflicting definitions found for name x in module __REPL
2: 0.to(100).exists(x => x == x)
                    ^
```

**Comment by ivan-gavran** [comment]
Thanks! Sure, I could change the name (though, in the example you gave, all should be changed to `y`, because we are looking for the integer from the range for which `fahrenheit(y) == y` , so in that scope different versions of the value do not mix. Don't know if that changes anything?)

Other than that, maybe it makes sense to mention in the docs that shadowing is not supported (probably nobody would make such a mistake in the actual spec, for readability, but in REPL people would do so, as I did).

Closing the issue
